server/src/main/java/org/opensearch/index/translog/transfer/TransferService.java

```diff
@@ -66,6 +66,23 @@ void uploadBlobs(
      */
     void uploadBlob(final TransferFileSnapshot fileSnapshot, Iterable<String> remotePath, WritePriority writePriority) throws IOException;
 
+    /**
+     * Reads the input stream and uploads as a blob
```

Comment:
Is there a limit on the data from the stream? does it needs uploaded as single blob?